.github/workflows/build.yml

```diff
@@ -18,17 +18,19 @@ jobs:
         with:
           python-version: 3.9
 
+      - name: Install poetry
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: "1.7.1"
```

Comment:
I believe it's better to not restrict specific version of package manager since this project should be compatible with the latest versions and also it will follow fail-fast approach